### ISSUE TITLE
Question about SARE and SFRS results (tested on tokyo247 dataset) used in the paper

### ISSUE THREAD

**Issue by hit2sjtu** [gmberton/CosPlace#12]
First I shall say really nice work.  I have some questions about Table 3 in your paper.
![image](https://user-images.githubusercontent.com/16236269/203460854-bf9d229e-b118-4dc5-a097-8f06b5d9815c.png)

For the SARE and SFRS model training, did you use the author's repo or your own re-implemented version (which I did not find in the codebase)? The model in the SFRS author's paper is also trained on pit30K and tested on tokyo247 dataset but the results are better. 
![image](https://user-images.githubusercontent.com/16236269/203461639-38596f45-48bf-4cf5-af91-d9aef9a4cf93.png)
I am wondering whether this is re-implemented side issue or due to "we followed deep learning's best practices (average over multiple runs for the main results, validation/early stopping and hyperparameter search on the val set)".
 It seems the SFRS author did pick the best model, as shown [https://github.com/yxgeee/OpenIBL/issues/2#issuecomment-686952506](https://github.com/yxgeee/OpenIBL/issues/2#issuecomment-686952506)

Thanks.

**Comment by gmberton** [comment]
Congrats on noticing the issue! You are right, the difference in results is due to the authors of SFRS picking the best model on the test set, as they admitted [here](https://github.com/yxgeee/OpenIBL/issues/9#issuecomment-724499053) as well.
So to get the results on our paper we used the [code of the authors](https://github.com/yxgeee/OpenIBL), following [all their recommendations](https://github.com/yxgeee/OpenIBL/blob/master/docs/INSTALL.md#pre-trained-weights) (i.e. starting from model weights from MatConvNet instead of Torchvision, and using the initial cluster centers for VLAD layer), and we performed a fair set of experiments (3 runs per experiment), without selecting the model on the test set.
